engine/runner/contracts/Proxy.sol

```diff
@@ -9,14 +9,19 @@ contract Proxy is FreeMarketBase, IHasUpstream {
   bytes32 constant runnerAddresses = 0x32b7d36eef9191cec628a9b46ddda74b702cf693ad48a065f3f9e5fcc4ea08f5; // keccak256('runnerAddresses')
 
   constructor(
-    address owner,
-    address storageAddress,
-    address upstream,
-    bool userProxy
-  ) FreeMarketBase(owner, storageAddress, upstream, userProxy) {}
+    address storageAddress
+  ) FreeMarketBase(storageAddress) {}
 
+  // TODO can we safely rename getUpstream() or upstreamAddress() so we have only 1?
   function getUpstream() external view virtual returns (address) {
-    return upstreamAddress;
+    return upstreamAddress();
```

Comment:
@0x-adam-smith can I remove upstreamAddress() or getUpstream()? they are duplicate methods. upstreamAddress() is tested in FrontDoor.ts and getUpstream() is required by IHasUpstream interface